test/extensions/filters/common/rbac/matchers_test.cc

```diff
@@ -624,6 +626,336 @@ TEST(UriTemplateMatcher, NoPathInHeader) {
+  EXPECT_CALL(info, route()).WillRepeatedly(Return(route));
+
+  // Test DYNAMIC source metadata match
+  {
+    envoy::type::matcher::v3::MetadataMatcher matcher;
+    matcher.set_filter("rbac");
+    matcher.add_path()->set_key("dynamic_key");
+    matcher.mutable_value()->mutable_string_match()->set_exact("dynamic_value");
+
+    checkMatcher(MetadataMatcher(Matchers::MetadataMatcher(matcher, context),
+                                 envoy::config::rbac::v3::MetadataSource::DYNAMIC),
+                 true, conn, header, info);
+
+    // Should not match with wrong value
+    matcher.mutable_value()->mutable_string_match()->set_exact("wrong_value");
+    checkMatcher(MetadataMatcher(Matchers::MetadataMatcher(matcher, context),
+                                 envoy::config::rbac::v3::MetadataSource::DYNAMIC),
+                 false, conn, header, info);
+  }
+
+  // Test ROUTE source metadata match
+  {
+    envoy::type::matcher::v3::MetadataMatcher matcher;
+    matcher.set_filter("rbac");
+    matcher.add_path()->set_key("route_key");
+    matcher.mutable_value()->mutable_string_match()->set_exact("route_value");
+
+    checkMatcher(MetadataMatcher(Matchers::MetadataMatcher(matcher, context),
+                                 envoy::config::rbac::v3::MetadataSource::ROUTE),
+                 true, conn, header, info);
+
+    // Should not match with wrong value
+    matcher.mutable_value()->mutable_string_match()->set_exact("wrong_value");
+    checkMatcher(MetadataMatcher(Matchers::MetadataMatcher(matcher, context),
+                                 envoy::config::rbac::v3::MetadataSource::ROUTE),
+                 false, conn, header, info);
+  }
+
+  // Test ROUTE source with null route
+  {
+    EXPECT_CALL(info, route()).WillRepeatedly(Return(nullptr));
+
+    envoy::type::matcher::v3::MetadataMatcher matcher;
+    matcher.set_filter("rbac");
+    matcher.add_path()->set_key("route_key");
+    matcher.mutable_value()->mutable_string_match()->set_exact("route_value");
+
+    checkMatcher(MetadataMatcher(Matchers::MetadataMatcher(matcher, context),
+                                 envoy::config::rbac::v3::MetadataSource::ROUTE),
+                 false, conn, header, info);
+  }
+}
+
+TEST(PortRangeMatcher, EdgeCases) {
+  Envoy::Network::MockConnection conn;
+  Envoy::Http::TestRequestHeaderMapImpl headers;
+  NiceMock<StreamInfo::MockStreamInfo> info;
+
+  // Test boundary conditions
+  envoy::type::v3::Int32Range range;
+  range.set_start(1);
+  range.set_end(65535);
+
+  // Test port at start of range
+  Envoy::Network::Address::InstanceConstSharedPtr addr1 =
+      Envoy::Network::Utility::parseInternetAddressNoThrow("1.2.3.4", 1, false);
+  info.downstream_connection_info_provider_->setLocalAddress(addr1);
+  checkMatcher(PortRangeMatcher(range), true, conn, headers, info);
+
+  // Test port at end of range - 1
+  Envoy::Network::Address::InstanceConstSharedPtr addr2 =
```

Comment:
Seems these are unreleated to this PR?

Reply:
Yeah, I was trying to increase the coverage of this class. I'll move it to a separate PR